### DISCUSSION TITLE
issue: useFieldArray doesn't retain provided values when appending

### DISCUSSION THREAD

**Discussion by abnerey** [react-hook-form/react-hook-form#12040] [Q&A]
### Version Number

^7.30.0

### Codesandbox/Expo snack

https://stackblitz.com/edit/vitejs-vite-qcckuk

### Steps to reproduce

1. Go to [https://stackblitz.com/edit/vitejs-vite-qcckuk](url)
2. See how the first two rows have changed their uniqueId (which is the keyName)
3. Add new rows
4. See how uniqueIds get overwritten by RHF 

### Expected behaviour

I saw that from version 7.23.0 ids are retained by RHF, but this only works for the default keyName, I consider that it makes sense to retain values on the provided keyName too so overrides can be avoided.

![image (1)](https://github.com/react-hook-form/react-hook-form/assets/11546757/b2780e0b-2587-4e80-b446-fd45431589b3)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
use `keyName` to change the `id`

**Reply by abnerey** [reply]
What I'm looking for is to `useFieldArray` to preserve the original values of whatever `keyName` was given, either default or custom, that's how it works currently when you use the default one (refer to the screenshot)

And the thing is that in the project I have we need to control the keyName outside of RHF

If it breaks something it can be a useFieldArray config property to avoid breaking changes or unexpected behaviours for current lib consumers
